src/app/tabs/compileTab/compileTab.js

```diff
@@ -46,14 +46,16 @@ class CompileTab {
     this.compiler.setEvmVersion(this.evmVersion)
   }
 
+  setLanguage (lang) {
+    this.compiler.setLanguage(lang)
+  }
+
   /**
    * Compile a specific file of the file manager
    * @param {string} target the path to the file to compile
    */
   compileFile (target) {
     if (!target) throw new Error('No target provided for compiliation')
-    if (!/\.sol/.exec(target)) throw new Error(`${target} is not a solidity file. It cannot be compiled with solidity compiler`)
-    // only compile *.sol file.
```

Comment:
should we check for .sol or .yul?

Reply:
There is a case (when we load a swarm hash) where the extension is not specified, but the content is still solidity. So it kinda make sense to try compiling it.
That basically why I removed that one.